File: DragableRecyclerView/src/main/java/kim/uno/sample/dragablerecyclerview/MainActivity.java
package kim.uno.sample.dragablerecyclerview;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;

import kim.uno.kotlin.base.ui.BaseRecyclerView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BaseRecyclerView recyclerSample = (BaseRecyclerView) findViewById(R.id.recycler_sample);
        recyclerSample.setScrollToTopButton(findViewById(R.id.v_top));
        recyclerSample.setLayoutManager(new LinearLayoutManager(this));

        SampleAdapter adapter = new SampleAdapter();
        recyclerSample.setAdapter(adapter);

        for (int i = 0; i < 20; i++) {
            Sample item = new Sample();
            item.isSwapable = i % 5 != 0;
            item.message = item.isSwapable? "swapable position " + i : "stone position " + i;
            adapter.addItem(item);
        }

        adapter.notifyDataSetChanged();
    }

}

File: Retrofit/src/main/java/kim/uno/sample/retrofit/MainActivity.kt
package kim.uno.sample.retrofit

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import io.reactivex.schedulers.Schedulers
import kim.uno.kotlin.base.util.LogUtil
import kim.uno.kotlin.base.util.ToastUtil
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var sampleService: SampleService? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        LogUtil.logEnable = true

        initRetrofit()
    }

    fun initRetrofit() {
        bt_retrofit.setOnClickListener { requestByRetrofit() }
    }

    private fun requestByRetrofit() {

        if (sampleService == null) sampleService = SampleService.retrofit.create(SampleService::class.java)

        sampleService!!.get("test")
                .subscribeOn(Schedulers.io())
                .subscribe { ToastUtil.show(this, "${it.message}") }
    }

}
